src/infra/src/table/migration/m20241213_133700_populate_destinations_table.rs

```diff
@@ -0,0 +1,333 @@
+    pub enum Relation {}
+
+    impl ActiveModelBehavior for ActiveModel {}
+}
+
+/// Representation of the destinations table at the time this migration executes.
+mod destinations {
+
+    use std::collections::HashMap;
+
+    use sea_orm::entity::prelude::*;
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
+    #[sea_orm(table_name = "destinations")]
+    pub struct Model {
+        #[sea_orm(primary_key, auto_increment = false)]
+        pub id: String,
+        pub org: String,
+        pub name: String,
+        pub module: String,
+        pub template_id: Option<String>,
+        pub pipeline_id: Option<String>,
+        pub r#type: Json,
+    }
+
+    // There are relations but they are not important to this migration.
+    #[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
+    pub enum Relation {}
+
+    impl ActiveModelBehavior for ActiveModel {}
+
+    #[derive(Clone, Debug, Serialize, Deserialize)]
+    #[serde(rename_all = "snake_case")]
+    pub struct Destination {
+        pub id: String,
+        pub org_id: String,
+        pub name: String,
+        pub module: Module,
+    }
+
+    #[derive(Serialize, Debug, Deserialize, Clone)]
+    #[serde(rename_all = "snake_case")]
+    pub enum Module {
+        Alert {
+            template_id: String,
+            destination_type: DestinationType,
+        },
+        Pipeline {
+            pipeline_id: String,
+            endpoint: Endpoint,
+        },
+    }
+
+    #[derive(Serialize, Debug, Deserialize, Clone)]
+    #[serde(tag = "type")]
+    #[serde(rename_all = "snake_case")]
+    pub enum DestinationType {
+        Http(Endpoint),
+        Email(Email),
+        Sns(AwsSns),
+    }
+
+    #[derive(Clone, Debug, Serialize, Deserialize)]
+    pub struct Email {
+        pub recipients: Vec<String>,
+    }
+
+    #[derive(Serialize, Debug, PartialEq, Eq, Deserialize, Clone)]
+    pub struct Endpoint {
+        pub url: String,
+        #[serde(default)]
+        pub method: HTTPType,
+        #[serde(default)]
+        pub skip_tls_verify: bool,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub headers: Option<HashMap<String, String>>,
+    }
+
+    #[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+    #[serde(rename_all = "snake_case")]
+    pub enum HTTPType {
+        #[default]
+        Post,
+        Put,
+        Get,
+    }
+
+    impl std::fmt::Display for HTTPType {
+        fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+            match self {
+                HTTPType::Post => write!(f, "post"),
+                HTTPType::Put => write!(f, "put"),
+                HTTPType::Get => write!(f, "get"),
+            }
+        }
+    }
+
+    #[derive(Clone, Debug, Serialize, Deserialize)]
+    pub struct AwsSns {
+        pub sns_topic_arn: String,
+        pub aws_region: String,
+    }
+}
```

Comment:
It looks like these structs aren't actually referenced/used anywhere?

I assume these are the "Service Layer" models? It makes sense to me that we don't actually use these since the migration just transforms records from the `meta` table into the SeaORM `destinations::Model`. I'm guessing these were just copied over at some point and never deleted?